crates/primitives/src/test_utils.rs

```diff
@@ -50,38 +53,33 @@ pub(crate) fn generate_pubkey_table(table: &[PublicKey]) -> PublickeyTable {
     PublickeyTable::from(pubkey_table)
 }
 
+pub(crate) fn generate_xonly_pubkey() -> XOnlyPublicKey {
+    let mut rng = thread_rng();
+    let mut sk = SecretKey::new(&mut rng);
+    // negate sk if odd
+    if sk.x_only_public_key(SECP256K1).1 == Parity::Odd {
```

Comment:
There should be an `EvenSecretKey` type, I think.

Reply:
Agreed, I think @Zk2u has something like this somewhere. But out of the scope of the PR.

Reply:
It's already there, just need to use it here:

```suggestion
    let even_sk: EvenSecretKey = sk.into();
```